include/grpc/grpc.h

```diff
@@ -170,14 +170,15 @@ void grpc_byte_buffer_destroy(grpc_byte_buffer *byte_buffer);
 struct grpc_byte_buffer_reader;
 typedef struct grpc_byte_buffer_reader grpc_byte_buffer_reader;
 
-grpc_byte_buffer_reader *grpc_byte_buffer_reader_create(
-    grpc_byte_buffer *buffer);
+/** Initialize \a reader to read over \a buffer */
+void grpc_byte_buffer_reader_init(grpc_byte_buffer_reader *reader,
+                                  grpc_byte_buffer *buffer);
+
 /* At the end of the stream, returns 0. Otherwise, returns 1 and sets slice to
    be the returned slice. Caller is responsible for calling gpr_slice_unref on
    the result. */
 int grpc_byte_buffer_reader_next(grpc_byte_buffer_reader *reader,
                                  gpr_slice *slice);
-void grpc_byte_buffer_reader_destroy(grpc_byte_buffer_reader *reader);
```

Comment:
Right... so:

grpc_byte_buffer_reader r;
grpc_byte_buffer_reader_init(&r, ...);
// do some wonderful things
grpc_byte_buffer_reader_destroy(&r);

On Mon, Jun 1, 2015 at 9:51 PM David G. Quintas <notifications@github.com>
wrote:

> In include/grpc/grpc.h
> <https://github.com/grpc/grpc/pull/1866#discussion_r31492707>:
>
> >  /* At the end of the stream, returns 0. Otherwise, returns 1 and sets slice to
> >     be the returned slice. Caller is responsible for calling gpr_slice_unref on
> >     the result. */
> >  int grpc_byte_buffer_reader_next(grpc_byte_buffer_reader *reader,
> >                                   gpr_slice *slice);
> > -void grpc_byte_buffer_reader_destroy(grpc_byte_buffer_reader *reader);
>
> The idea being to encapsulate the creation/destruction, as opposed to
> having the user gpr_{malloc,free} "in the open"?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc/pull/1866/files#r31492707>.
>